### ISSUE TITLE
Can't register KeePassHTTP extensions keys on Archlinux

### ISSUE THREAD

**Issue by Generator** [keepassxreboot/keepassxc#724]
## Expected Behavior

Install new KeePassHTTP browser extension (e.g. PassiFox, keepassxc-browser, KeePassHelper)

## Current Behavior

Unable to register new KeePassHTTP extension key.
There's no prompt to register the new extension key.

Tested with official KeePassXC-2.2.0-x86_64.AppImage and works without problem.

## Steps to Reproduce (for bugs)
Install PassiFox or other KeePassHTTP extension, register new extension KeePassHTTP key.

## Debug Info

[KeePassXC](https://www.archlinux.org/packages/community/x86_64/keepassxc/) - 2.2.0

Libraries:
- [qt5-x11extras](https://www.archlinux.org/packages/extra/x86_64/qt5-x11extras/) 5.9.0
- [qt5-tools](https://www.archlinux.org/packages/extra/x86_64/qt5-tools/) 5.9.0
- [libxtst](https://www.archlinux.org/packages/extra/x86_64/libxtst/) 1.2.3

Operating system: Archlinux
CPU architecture: X86_64
Kernel: 4.11.7-1-ck-core2

Enabled extensions:
- KeePassHTTP

Issue on Archlinux: https://bugs.archlinux.org/task/54599

**Comment by phoerious** [comment]
I'm not quite sure what you bug is. Can you provide some more detailed reproduction steps?

**Comment by Generator** [comment]
@phoerious 
Do a clean install of KeePassXC on Archlinux and enable KeePassHTTP, Install a browser extension (Chrome or Firefox) should appear a dialogue to accept the new extension key to access KeePassXC through KeePassHTTP.

I've tested with the archlinux package and the dialogue didn't show, also tested with Appimage package with no issue registering the extension key.  

Reported the issue on Archlinux, was tell me to report on upstream.

**Comment by phoerious** [comment]
Works perfectly fine for me.

1. Install chromeIPass
2. Start KeePassXC
3. Enable KeePassHTTP in the settings
4. Click "Connect" in chromeIPass
5. Enter a name
6. Profit

**Comment by Generator** [comment]
Look like there was some issue with previews configuration files.
Removed $HOME/.config/keepassxc/ and start KeePassXC again solved the problem.

**Comment by purejava** [comment]
Tried KeePassXC for the first time on current Arch linux

- FF 58.0.2 (64bit)
- KeePassXC 2.2.4
- KeePassXC-Browser Version: 0.6.0

I can't connect to the database. The FF error console shows a lot of errors:

> b1519547384606 addons.webextension.jid1-f3mYMbCpz2AZYl@jetpack WARN  Please specify whether you want browser_style or not in your browser_action options.
1519547384607 addons.webextension.keepassxc-browser@keepassxc.org WARN  Please specify whether you want browser_style or not in your browser_action options.
1519547384607 addons.webextension.firefox@ghostery.com  WARN  Please specify whether you want browser_style or not in your browser_action options.
1519547384607 addons.webextension.{bee6eb20-01e0-ebd1-da83-080329fb9a3a}  WARN  Please specify whether you want browser_style or not in your browser_action options.
Synchrone XMLHttpRequests am Haupt-Thread sollte nicht mehr verwendet werden, weil es nachteilige Effekte für das Erlebnis der Endbenutzer hat. Für weitere Hilfe siehe http://xhr.spec.whatwg.org/  browser-core.js:3156:2
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src moz-extension://fe037dd5-2332-4d1d-9313-2d2766cffd9a"). Source: call to eval() or related function blocked by CSP.  vendor.js:12694
Connecting to native messaging host org.keepassxc.keepassxc_browser  keepass.js:870:5
1519547386069 Toolkit.Telemetry WARN  TelemetryEnvironment::setExperimentActive - the experiment type was truncated.
No such native application org.keepassxc.keepassxc_browser  NativeMessaging.jsm:100
Failed to connect: Unknown error  keepass.js:846:5
Tastenereignis ist in manchen Tastaturlayouts nicht verfügbar: Taste="r" Modifikatoren="accel,alt" ID=“key_toggleReaderMode”  browser.xul
Tastenereignis ist in manchen Tastaturlayouts nicht verfügbar: Taste="i" Modifikatoren="accel,alt,shift" ID=“key_browserToolbox”  browser.xul
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
Connecting to native messaging host org.keepassxc.keepassxc_browser  keepass.js:870:5
No such native application org.keepassxc.keepassxc_browser  NativeMessaging.jsm:100
  this.NativeApp</this.startupPromise< resource://gre/modules/NativeMessaging.jsm:100:9
Failed to connect: Unknown error  keepass.js:846:5
Connecting to native messaging host org.keepassxc.keepassxc_browser  keepass.js:870:5
No such native application org.keepassxc.keepassxc_browser  NativeMessaging.jsm:100
Failed to connect: Unknown error  keepass.js:846:5
[Exception... "Component returned failure code: 0xc1f30001 (NS_ERROR_NOT_INITIALIZED) [nsIMessageSender.sendAsyncMessage]"  nsresult: "0xc1f30001 (NS_ERROR_NOT_INITIALIZED)"  location: "JS frame :: resource://gre/modules/ExtensionUtils.jsm :: sendAsyncMessage :: line 514"  data: no]  (unbekannt)
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
TypeError: keepass.nativePort is null[Weitere Informationen]  keepass.js:92:13
uncaught exception: false  (unbekannt)
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
TypeError: keepass.nativePort is null[Weitere Informationen]  keepass.js:92:13
uncaught exception: false  (unbekannt)
Discarding onCreatedNavigationTarget for 67: received source tab data without any created tab data available
Unknown label for categorical histogram  (unbekannt)
Discarding onCreatedNavigationTarget for 67: received source tab data without any created tab data available
Invalid chrome URI: /
[Exception... "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIIOService.newURI]"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource://gre/modules/NetUtil.jsm :: NetUtil_newURI :: line 209"  data: no]  (unbekannt)
Discarding onCreatedNavigationTarget for 67: received source tab data without any created tab data available
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
Connecting to native messaging host org.keepassxc.keepassxc_browser  keepass.js:870:5
No such native application org.keepassxc.keepassxc_browser  NativeMessaging.jsm:100
  this.NativeApp</this.startupPromise< resource://gre/modules/NativeMessaging.jsm:100:9
Failed to connect: Unknown error  keepass.js:846:5
Connecting to native messaging host org.keepassxc.keepassxc_browser  keepass.js:870:5
No such native application org.keepassxc.keepassxc_browser  NativeMessaging.jsm:100
Failed to connect: Unknown error  keepass.js:846:5
[Exception... "Component returned failure code: 0xc1f30001 (NS_ERROR_NOT_INITIALIZED) [nsIMessageSender.sendAsyncMessage]"  nsresult: "0xc1f30001 (NS_ERROR_NOT_INITIALIZED)"  location: "JS frame :: resource://gre/modules/ExtensionUtils.jsm :: sendAsyncMessage :: line 514"  data: no]  (unbekannt)
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
TypeError: keepass.nativePort is null[Weitere Informationen]  keepass.js:92:13
uncaught exception: false  (unbekannt)
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
TypeError: keepass.nativePort is null[Weitere Informationen]  keepass.js:92:13
uncaught exception: false  (unbekannt)
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
Connecting to native messaging host org.keepassxc.keepassxc_browser  keepass.js:870:5
No such native application org.keepassxc.keepassxc_browser  NativeMessaging.jsm:100
Failed to connect: Unknown error  keepass.js:846:5
[Exception... "Component returned failure code: 0xc1f30001 (NS_ERROR_NOT_INITIALIZED) [nsIMessageSender.sendAsyncMessage]"  nsresult: "0xc1f30001 (NS_ERROR_NOT_INITIALIZED)"  location: "JS frame :: resource://gre/modules/ExtensionUtils.jsm :: sendAsyncMessage :: line 514"  data: no]  (unbekannt)
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
TypeError: keepass.nativePort is null[Weitere Informationen]  keepass.js:92:13
uncaught exception: false  (unbekannt)
<unavailable>  keepassxc-browser.js:1242
Error: Promised response from onMessage listener at init@moz-extension://fe037dd5-2332-4d1d-9313-2d2766cffd9a/dist/browser-core.js:17566:5 went out of scope  undefined
<unavailable>  keepassxc-browser.js:1242
Error: Promised response from onMessage listener at init@moz-extension://fe037dd5-2332-4d1d-9313-2d2766cffd9a/dist/browser-core.js:17566:5 went out of scope  undefined
Error 5: Timeout or not connected to KeePassXC  keepass.js:954:5
Discarding onCreatedNavigationTarget for 67: received source tab data without any created tab data available

**Comment by phoerious** [comment]
You need KeePassXC 2.3.

**Comment by purejava** [comment]
Thanks for the hint.
I compiled and installed [2.3.0-beta1](https://github.com/keepassxreboot/keepassxc/releases), but this did not change anything.
![xc](https://user-images.githubusercontent.com/1822238/36641249-cb2eea18-1a2c-11e8-89c6-6a130695bc41.jpg)

**Comment by phoerious** [comment]
You can just download the beta from here: https://github.com/keepassxreboot/keepassxc/releases/tag/2.3.0-beta1

**Comment by purejava** [comment]
There is no icon to activate my browser. Only legacy ...

![xc1](https://user-images.githubusercontent.com/1822238/36641592-5284f5c0-1a32-11e8-8b93-a10e68406767.png)

**Comment by phoerious** [comment]
You compiled without -DWITH_XC_BROWSER=ON.

**Comment by purejava** [comment]
Now we are talking! :) I replaced `-DWITH_XC_HTTP=ON` in the PKGBUILD with `-DWITH_XC_BROWSER=ON` and the icon showed up. I could connect and KeePassXC is working now.

**Thank you very much for your quick and continous support!!**

![xc2](https://user-images.githubusercontent.com/1822238/36641959-92c3c46c-1a38-11e8-8f5f-884a03ea9c0e.jpg)
